Feature/rds utils

alichherawalla opened the pull request:
#### Description of changes
This adds support for $util.rds.toJsonString

#### Description of how you validated changes
manual testing and `yarn test`


#### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)


By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

alichherawalla commented:
Hi @cjihrig, 

Is this something that you could help with?

cjihrig commented on packages/amplify-appsync-simulator/src/__tests__/velocity/util/rds.test.ts:
```diff
@@ -0,0 +1,23 @@
+
+const stubInfo = {} as unknown;
+export const mockInfo = stubInfo as GraphQLResolveInfo;
+var util;
```
```suggestion
let util;
```

cjihrig commented on packages/amplify-appsync-simulator/src/__tests__/velocity/util/rds.test.ts:
```diff
@@ -0,0 +1,23 @@
+import { GraphQLResolveInfo } from 'graphql';
+
+const stubInfo = {} as unknown;
+export const mockInfo = stubInfo as GraphQLResolveInfo;
```
Why is this exported?

cjihrig reviewed (commented):
Left a few comments. The code generally looks good to me, but I'd like to test it a bit.

alichherawalla commented:
> Left a few comments. The code generally looks good to me, but I'd like to test it a bit.

thanks for the super fast turnaround, I have addressed the comments. 

Thank you and I'll wait to hear back from you after your testing

cjihrig commented on packages/amplify-appsync-simulator/src/velocity/util/rds.ts:
```diff
@@ -0,0 +1,19 @@
+  toJsonString: rdsObject => {
+    try {
+      rdsObject = JSON.parse(rdsObject);
+      const rdsJson = (rdsObject?.sqlStatementResults || []).map(statement =>
```
```suggestion
      const rdsJson = (rdsObject?.sqlStatementResults ?? []).map(statement =>
```

cjihrig commented on packages/amplify-appsync-simulator/src/velocity/util/rds.ts:
```diff
@@ -0,0 +1,19 @@
+    try {
+      rdsObject = JSON.parse(rdsObject);
+      const rdsJson = (rdsObject?.sqlStatementResults || []).map(statement =>
+        (statement?.records || []).map(record => {
```
```suggestion
        (statement?.records ?? []).map(record => {
```

cjihrig commented on packages/amplify-appsync-simulator/src/velocity/util/rds.ts:
```diff
@@ -0,0 +1,19 @@
+        (statement?.records || []).map(record => {
+          const result = {};
+          record.forEach((row, index) => {
+            result[statement.columnMetadata[index].name] = Object.values(row)[0];
```
```suggestion
            result[statement?.columnMetadata?.[index]?.name] = Object.values(row)?.[0];
```

alichherawalla commented:
Done, thank you for the suggestions

cjihrig reviewed (approved):
LGTM, thanks!
